web/src/pages/ForgotPassword/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useCallback, useRef, useContext } from 'react';
+import { Link } from 'react-router-dom';
+
+import { Container } from '../SignUp/styles';
+import { FiMail, FiAtSign } from 'react-icons/fi';
+import Input from '../../components/Input';
+import { Form } from '@unform/web';
+import { FormHandles } from '@unform/core';
+import * as Yup from 'yup';
+import getValidationErrors from '../../utils/getValidationErrors';
+import { AnimationContainer } from '../SignIn/styles';
+import api from '../../services/api';
+import { ToastContext } from '../../context/ToastContext';
+
+interface ForgotPasswordFormData {
+  email: string;
+  password: string;
+}
+
+const ForgotPassword: React.FC = () => {
+  const formRef = useRef<FormHandles>(null);
+  const { addToast } = useContext(ToastContext);
+
+  const handleSubmit = useCallback(async (data: ForgotPasswordFormData) => {
+    try {
+      formRef.current?.setErrors({});
+
+      const schema = Yup.object().shape({
+        email: Yup.string()
+          .required('Email é obrigatório')
+          .email('Digite um email válido'),
+      });
+
+      await schema.validate(data, {
+        abortEarly: false,
+      });
+
+      await api.post('/users/reset-password', data);
+
+      addToast({
```

Comment:
Toasts won't appear for me. I tried changing `hasdescription` to a string (`'true'|'false'`) as the error message suggested, to no avail.

Reply:
That has never occurred to me, I'll check that later

Reply:
Unrelated to PR + will get obliterated by React 18 + []'s